core/change_test.go

```diff
@@ -331,20 +331,102 @@ var _ = Describe("Change ask prompts", func() {
 		}()
 
 		c := &Change{}
-		Expect(AskPrompts(c, config, stdinReader)).NotTo(Succeed())
+		Expect(c.AskPrompts(config, stdinReader)).NotTo(Succeed())
 	})
 
-	It("panics when trying to find kind from bad label", func() {
+	It("errors when trying to find kind from bad label", func() {
 		config := Config{
 			Kinds: []KindConfig{
-				{Label: "a"},
+				{Label: "kind"},
+			},
+		}
+
+		c := &Change{
+			Kind: "not kind",
+			Body: "body",
+		}
+
+		err := c.AskPrompts(config, stdinReader)
+		Expect(err).NotTo(BeNil())
+		Expect(errors.Unwrap(err)).To(Equal(errInvalidKind))
+	})
+
+	It("doesn't prompt for kind if it's already set", func() {
+		config := Config{
+			Kinds: []KindConfig{
+				{Label: "kind"},
+			},
+		}
+
+		c := &Change{
+			Kind: "kind",
+		}
+
+		go func() {
+			DelayWrite(stdinWriter, []byte("body"))
+			DelayWrite(stdinWriter, []byte{13})
+		}()
+
+		Expect(c.AskPrompts(config, stdinReader)).To(Succeed())
+		Expect(c.Kind).To(Equal("kind"))
+		Expect(c.Body).To(Equal("body"))
+	})
+
+	It("doesn't prompt for body if it's already set", func() {
+		config := Config{
+			Kinds: []KindConfig{
+				{Label: "kind"},
+			},
+		}
+
+		c := &Change{
+			Body: "body",
+		}
+
+		go func() {
+			DelayWrite(stdinWriter, []byte("kind"))
+			DelayWrite(stdinWriter, []byte{13})
+		}()
+
+		Expect(c.AskPrompts(config, stdinReader)).To(Succeed())
+		Expect(c.Kind).To(Equal("kind"))
+		Expect(c.Body).To(Equal("body"))
+	})
+
+	It("errors when body is given for a kind that shouldn't have one", func() {
+		config := Config{
+			Kinds: []KindConfig{
+				{Label: "kind", SkipBody: true},
+			},
+		}
+
+		c := &Change{
+			Body: "body",
+			Kind: "kind",
+		}
+
+		err := c.AskPrompts(config, stdinReader)
+		Expect(err).To(Equal(errKindDoesNotAcceptBody))
+	})
+
+	It("validates body without prompt", func() {
+		var min int64 = 10
+
+		config := Config{
+			Kinds: []KindConfig{
+				{Label: "kind"},
+			},
+			Body: BodyConfig{
+				MinLength: &min,
 			},
 		}
-		badKindFromLabel := func() {
-			_ = kindFromLabel(config, "not a")
-			Fail("should panic before this")
+
+		c := &Change{
+			Kind: "kind",
+			Body: "body",
 		}
-		defer GinkgoRecover()
-		Expect(badKindFromLabel).To(Panic())
+
+		err := c.AskPrompts(config, stdinReader)
+		Expect(errors.Unwrap(err)).To(Equal(errInputTooShort))
```

Comment:
```suggestion
		Expect(errors.Is(err)).To(Equal(errInputTooShort))
```
I think using errors.is is better here, us wrapping the error shouldn't matter.